banzai_floyds/utils/binning_utils.py

```diff
@@ -0,0 +1,63 @@
+from astropy.table import Table, vstack
+import numpy as np
+
+
+def bin_data(data, uncertainty, wavelengths, orders, mask=None):
+    if mask is None:
+        mask = np.zeros_like(data, dtype=int)
+    binned_data = None
+    x2d, y2d = np.meshgrid(np.arange(data.shape[1]), np.arange(data.shape[0]))
+    for order_id, order_bins in zip(orders.order_ids, wavelengths.bin_edges):
+        in_order = orders.data == order_id
+
+        y_order = y2d[in_order] - orders.center(x2d[in_order])[order_id - 1]
+        data_table = Table({'data': data[in_order], 'uncertainty': uncertainty[in_order],
+                            'mask': mask[in_order], 'wavelength': wavelengths.data[in_order],
+                            'x': x2d[in_order], 'y': y2d[in_order], 'y_order': y_order})
+        bin_number = np.digitize(data_table['wavelength'], order_bins)
+        bin_centers = (order_bins[1:] + order_bins[:-1]) / 2.0
+        # Append the first and last bin centers as zero to flag that the
+        # edge pixels aren't in a bin
+        bin_centers = np.hstack([0, bin_centers, 0])
+        bin_widths = (order_bins[1:] - order_bins[:-1])
+        bin_widths = np.hstack([0, bin_widths, 0])
+        data_table['wavelength_bin'] = bin_centers[bin_number]
+        data_table['wavelength_bin_width'] = bin_widths[bin_number]
+        data_table['order'] = order_id
+        if binned_data is None:
+            binned_data = data_table
+        else:
+            binned_data = vstack([binned_data, data_table])
+    return binned_data.group_by(('order', 'wavelength_bin'))
+
+
+def combine_wavelength_bins(wavelength_bins):
```

Comment:
This is only a refactor, but I'm pretty sure this logic is wrong. It is not used yet and will be the subject of a future PR.